presto-hive/src/main/java/com/facebook/presto/hive/HiveSplit.java

```diff
@@ -47,7 +48,8 @@
     private final OptionalInt readBucketNumber;
     private final OptionalInt tableBucketNumber;
     private final boolean forceLocalScheduling;
-    private final Map<Integer, HiveType> columnCoercions; // key: hiveColumnIndex
+    private final int partitionDataColumnCount;
+    private final Map<Integer, Column> partitionSchemaDifference; // key: hiveColumnIndex
```

Comment:
nit: What about name this `partitionSchmeaOverride` ? Ditto for other places.

Reply:
Than it should be something like `partitionColumnSchemaOverrides`. As this doesn't override the whole partition schema, but only some columns. Thus i like the `partitionSchemaDifference` more. I would like to keep it that way if you don't mind.